iosapp/lib-ios/Sources/LibApp/IOSReducer.swift

```diff
@@ -0,0 +1,924 @@
+      ) {
+        self.firstLaunch = firstLaunch
+        self.batteryLevel = batteryLevel
+        self.majorOnboarder = majorOnboarder
+        self.ownsMac = ownsMac
+        self.returningTo = returningTo
+        self.availableDiskSpaceInBytes = availableDiskSpaceInBytes
+        self.startClearCache = startClearCache
+        self.endClearCache = endClearCache
+      }
+
+      mutating func takeReturningTo() -> IOSReducer.Screen? {
+        let returningTo = self.returningTo
+        self.returningTo = nil
+        return returningTo
+      }
+    }
+  }
+
+  @ObservationIgnored
+  @Dependency(\.api) var api
+  @ObservationIgnored
+  @Dependency(\.appStore) var appStore
+  @ObservationIgnored
+  @Dependency(\.device) var device
+  @ObservationIgnored
+  @Dependency(\.filter) var filter
+  @ObservationIgnored
+  @Dependency(\.systemExtension) var systemExtension
+  @ObservationIgnored
+  @Dependency(\.storage) var storage
+  @ObservationIgnored
+  @Dependency(\.date.now) var now
+  @ObservationIgnored
+  @Dependency(\.locale) var locale
+  @ObservationIgnored
+  @Dependency(\.mainQueue) var mainQueue
+
+  enum CancelId {
+    case cacheClearUpdates
+  }
+
+  public init() {}
+
+  public var body: some Reducer<State, Action> {
+    Reduce { state, action in
+      switch action {
+      case .interactive(let interactiveAction):
+        return self.interactive(state: &state, action: interactiveAction)
+      case .programmatic(let programmaticAction):
+        return self.programmatic(state: &state, action: programmaticAction)
+      }
+    }
+  }
+
+  func interactive(state: inout State, action: Action.Interactive) -> Effect<Action> {
+    switch action {
+    case .onboardingBtnTapped(let btn, _):
+      return self.onboardingBtnTapped(btn, state: &state, action: action)
+
+    case .blockGroupToggled(let group):
+      self.log("block group toggled: \(group)", "02976f9b")
+      state.disabledBlockGroups.toggle(group)
+      return .none
+
+    case .sheetDismissed where state.screen == .onboarding(.major(.explainAppleFamily)):
+      self.log(state.screen, action, "118d259f")
+      state.screen = .onboarding(.major(.askIfInAppleFamily))
+      return .none
+
+    case .sheetDismissed:
+      return .none
+
+    case .receivedShake where state.screen.isRunning:
+      state.screen = .running(showVendorId: true)
+      return .run { send in
+        guard let disabled = self.storage.loadDisabledBlockGroups(),
+              let vendorId = self.device.vendorId else {
+          return
+        }
+        let rules = try await self.api.fetchBlockRules(
+          vendorId: vendorId,
+          disabledGroups: disabled
+        )
+        self.storage.saveProtectionMode(.normal(rules))
+        try await self.filter.notifyRulesChanged()
+      }
+
+    case .receivedShake where state.screen == .onboarding(.happyPath(.hiThere)):
+      #if DEBUG
+        state.screen = .onboarding(.happyPath(.optOutBlockGroups))
```

Comment:
i test this stuff dozens of times on my iphone, this lets me skip to the install area, which is often what i want to get to, as it takes a minute or two if i don't skip